src/useMemoCache/__docs__/example.stories.tsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useMemo, useRef, useState } from 'react';
+import { useMemoCache } from '../..';
+
+const dependencyListMapper = {
+  true: [1, 3],
+  false: [2, 4],
+};
+
+const getSummary = (numbers: number[]) => numbers.reduce((a, b) => a + b);
+
+const boolToString = (bool: boolean) => String(bool) as 'true' | 'false';
+
+export const Example: React.FC = () => {
+  const isTruthy = useRef(false);
+
+  const [dependencyList, setDependencyList] = useState(
+    () => dependencyListMapper[boolToString(isTruthy.current)]
+  );
+
+  const memoSpy = useRef(0);
+  const memoCacheSpy = useRef(0);
+
+  const memo = useMemo(() => {
+    memoSpy.current++;
+
+    return getSummary(dependencyList);
+  }, dependencyList);
+
+  const memoCache = useMemoCache(() => {
+    memoCacheSpy.current++;
+
+    return getSummary(dependencyList);
+  }, dependencyList);
+
+  const toggleDependencyList = () => {
+    isTruthy.current = !isTruthy.current;
+
+    setDependencyList(dependencyListMapper[boolToString(isTruthy.current)]);
+  };
+
+  return (
+    <div>
+      <section>
+        <h1>Memo</h1>
+        <p>summary: {memo}</p>
+        <p>calls: {memoSpy.current}</p>
+      </section>
+      <section>
+        <h1>Memo with cache</h1>
+        <p>summary: {memoCache}</p>
+        <p>calls: {memoCacheSpy.current}</p>
+      </section>
+      <button onClick={toggleDependencyList}>toggle dependency list</button>
+    </div>
+  );
+};
```

Comment:
The example is good, but I think the `summary` value just being toggled between 4 and 6 is really confusing. I think the example needs to do something a bit more realistic to really illustrate the functionality of the hook. Maybe you could do something like let the user input a number and then count all Fibonacci numbers until that point while displaying the number of calls made to the factory functions below.

Also, make sure that the example is understandable by using the example program and reading the example code. Storybook doesn't show all code that you have in the `example.stories.tsx` file. For example, now `getSummary` and `dependencyListMapper` are not visible in the example code, which prevented me from understanding the example when reading through Storybook.

Lastly, `getSummary` should be renamed to `sum`, if you intend to keep it.